mullvad-management-interface/src/lib.rs

```diff
@@ -122,7 +122,11 @@ pub async fn new_rpc_client() -> Result<ManagementServiceClient, Error> {
     // The URI will be ignored
     let channel = Endpoint::from_static("lttp://[::]:50051")
         .connect_with_connector(service_fn(move |_: Uri| {
-            IpcEndpoint::connect(ipc_path.clone())
+            let ipc_path = ipc_path.clone();
+            async {
+                let stream = IpcEndpoint::connect(ipc_path).await?;
+                Ok::<_, std::io::Error>(hyper_util::rt::tokio::TokioIo::new(stream))
+            }
```

Comment:
[TryFutureExt::map_ok](https://docs.rs/futures/latest/futures/future/trait.TryFutureExt.html#method.map_ok) might look nicer here

Reply:
Good idea, I could even remove the entire `async`-block.